### ISSUE TITLE
The results obtained during validation and test are inconsistent

### ISSUE THREAD

**Issue by dongzhetao** [XPixelGroup/BasicSR#675]
My validation set and my test set are the same, and when I use the same number of iterations of the network parameters to get my metrics, I find that they get different metrics

**Comment by dongzhetao** [comment]
For example, the psnr obtained after 100k validation is different from the psnr obtained when I test the model parameters saved by 100k

**Comment by sq-3768** [comment]
Hi, bro! I have the same issue too, have u solved it? The dataset is completely consistent and has not been enhanced or shuffled. The functions called by the code are also the same, but I don't know why the test results and validation results are inconsistent

> For example, the psnr obtained after 100k validation is different from the psnr obtained when I test the model parameters saved by 100k

**Comment by sq-3768** [comment]
> For example, the psnr obtained after 100k validation is different from the psnr obtained when I test the model parameters saved by 100k

I just solved this problem because the model enabled EMA mode during the training phase, but did not use EMA parameters during the testing phase, resulting in inconsistent validation and testing results. So the solution to the problem is to add "param_key_g: params_ema" to the path field in the yml file being tested
